src/components/MoveMediaModal/MoveMediaModal.tsx

```diff
@@ -0,0 +1,326 @@
+import {
+  Box,
+  HStack,
+  Icon,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Skeleton,
+  Text,
+  VStack,
+} from "@chakra-ui/react"
+import {
+  Button,
+  Infobox,
+  ModalCloseButton,
+} from "@opengovsg/design-system-react"
+import { AxiosError } from "axios"
+import _ from "lodash"
+import { useEffect, useState } from "react"
+import { BiArrowBack, BiFolderOpen } from "react-icons/bi"
+import { UseMutateFunction } from "react-query"
+import { useParams } from "react-router-dom"
+
+import { Breadcrumbs } from "components/Breadcrumbs"
+import { DirMenuItem, FileMenuItem } from "components/move/MoveMenuItem"
+
+import {
+  useListMediaFolderFiles,
+  useListMediaFolderSubdirectories,
+} from "hooks/directoryHooks"
+
+import { MiddlewareError } from "types/error"
+import {
+  MediaFolderTypes,
+  MediaLabels,
+  MoveSelectedMediaDto,
+  SelectedMediaDto,
+} from "types/media"
+
+interface MoveMediaModalProps {
+  selectedMedia: SelectedMediaDto[]
+  mediaType: MediaFolderTypes
+  mediaLabels: MediaLabels
+  isWriteDisabled: boolean | undefined
+  isOpen: boolean
+  isLoading: boolean
+  onClose: () => void
+  onProceed: UseMutateFunction<
+    void,
+    AxiosError<MiddlewareError>,
+    MoveSelectedMediaDto,
+    unknown
+  >
+}
+
+export const MoveMediaModal = ({
+  selectedMedia,
+  mediaType,
+  mediaLabels,
+  isWriteDisabled,
+  isOpen,
+  isLoading,
+  onClose,
+  onProceed,
+}: MoveMediaModalProps): JSX.Element => {
+  const { siteName } = useParams<{ siteName: string }>()
+  const { singularMediaLabel, pluralMediaLabel } = mediaLabels
+
+  const [moveTo, setMoveTo] = useState<string>(mediaType)
+
+  const {
+    data: mediaSubdirectories,
+    isLoading: isSubdirectoriesLoading,
+  } = useListMediaFolderSubdirectories({
+    siteName,
+    mediaDirectoryName: moveTo,
+  })
+
+  const {
+    data: mediaFilesData,
+    isLoading: isMediaFilesLoading,
+  } = useListMediaFolderFiles({
+    siteName,
+    mediaDirectoryName: moveTo,
+  })
+
+  const onModalClose = () => {
+    setMoveTo(mediaType)
+    onClose()
+  }
+
+  useEffect(() => {
+    // Note: This useEffect is needed to reset the state of the modal when it
```

Comment:
Hmm given that this entire modal gets re-rendered when isOpen changes, could we just put 
`setMoveTo(mediaType)` without the useEffect instead?

Reply:
Hmm no unfortunately not possible, because the modal is rendered immediately on mount. The `isOpen` variable only sets the `display: none` CSS style, but in the background the modal is already loaded.